packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -0,0 +1,140 @@
+import { shorthandBuckets, kebabCase, type ShorthandProperties } from '@compiled/utils';
+import type { Rule, Scope } from 'eslint';
+
+import { isStyled, isCss, isCssMap } from '../../index';
+
+const fixProperties = (context: Rule.RuleContext, node: Rule.Node) => {
```

Comment:
I am still not sold on the autofixer:
- it may cause unwanted side effects in existing code. Do we have an estimate of the current violations?
- a more descriptive error message should be enough

cc @dddlr @kylorhall-atlassian

Reply:
with the current approach, running the eslint rule in jira gives 4395 violations

though looking at the specific violations that were detected, it looks like this is far more violations than are actually necessary 🤔

the current approach [checks `shorthandBuckets`](https://github.com/atlassian-labs/compiled/blob/master/packages/utils/src/shorthand.ts#L526), which is a mapping of what "buckets" 
each property should fall into, such that more broad shorthand properties (`border`) always come before more specific shorthand properties (`borderColor`, `borderTop`), and more specific shorthand properties always come before longhand properties (`borderTopColor`).

for example, `border` is in bucket 1, while `borderColor` is in bucket 4, meaning that `border` should come before `borderColor`. so for the following code:

```tsx
const styles = css({
  borderColor: '...' // <-- bucket 4
  border: '...', // <-- bucket 1
});
```

the rule will correctly mark this as a lint violation, because a bucket 4 property erroneously comes before a bucket 1 property.

but if we have something like this:

```tsx
const styles = css({
  borderBottom: '...' // <-- bucket 4
  font: '...' // <-- bucket 1
});
```

the rule will still mark this as a lint violation, even though the properties have nothing to do with each other.

maybe an improvement we could do for the ESLint rule is to also check `shorthandFor` (or a custom variable based on `shorthandFor`) to ensure that

* properties that are shorthands/longhands of each other are in the right order (e.g. `border`, `borderColor`, `borderBottom`, `borderTopColor`), while
* properties that are not shorthands/longhands of each other do not trigger a lint violation?

the more lint violations, the more our team or jira developers have to fix... which is not ideal

Reply:
> the rule will still mark this as a lint violation, even though the properties have nothing to do with each other.
This doesn't seem right

> the more lint violations, the more our team or jira developers have to fix
agreed. However, I still believe we shouldn't have an autofixer. A suppression should be enough for existing cases so that developers can be educated and fix their own mistakes; a clear error message should be enough for instant feedback while introducing changes